src/main/java/net/staticsnow/nexus/repository/apt/AptUploadHandler.java

```diff
@@ -0,0 +1,103 @@
+import org.sonatype.nexus.repository.rest.UploadDefinitionExtension;
+import org.sonatype.nexus.repository.security.ContentPermissionChecker;
+import org.sonatype.nexus.repository.security.VariableResolverAdapter;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreBlob;
+import org.sonatype.nexus.repository.upload.ComponentUpload;
+import org.sonatype.nexus.repository.upload.UploadDefinition;
+import org.sonatype.nexus.repository.upload.UploadHandlerSupport;
+import org.sonatype.nexus.repository.upload.UploadResponse;
+
+import net.staticsnow.nexus.repository.apt.internal.AptFormat;
+import net.staticsnow.nexus.repository.apt.internal.AptPackageParser;
+import net.staticsnow.nexus.repository.apt.internal.FacetHelper;
+import net.staticsnow.nexus.repository.apt.internal.debian.ControlFile;
+import net.staticsnow.nexus.repository.apt.internal.hosted.AptHostedFacet;
+
+@Singleton
+@Named(AptFormat.NAME)
+public class AptUploadHandler
+    extends UploadHandlerSupport
+{
+  private final VariableResolverAdapter variableResolverAdapter;
+  private final ContentPermissionChecker contentPermissionChecker;
+  private UploadDefinition definition;
+  @Inject
+  public AptUploadHandler(@Named("simple") final VariableResolverAdapter variableResolverAdapter,
+                          final ContentPermissionChecker contentPermissionChecker,
+                          final Set<UploadDefinitionExtension> uploadDefinitionExtensions)
+  {
+    super(uploadDefinitionExtensions);
+    this.variableResolverAdapter = variableResolverAdapter;
+    this.contentPermissionChecker = contentPermissionChecker;
+  }
+
+  @Override
+  public UploadResponse handle(Repository repository, ComponentUpload upload) throws IOException {
+    AptHostedFacet hostedFacet = repository.facet(AptHostedFacet.class);
+    StorageFacet storageFacet = repository.facet(StorageFacet.class);
+    return TransactionalStoreBlob.operation.withDb(storageFacet.txSupplier()).throwing(IOException.class).call(() -> {
+      try (TempBlob tempBlob = storageFacet.createTempBlob(upload.getAssetUploads().get(0).getPayload(), FacetHelper.hashAlgorithms)) {
+        
+        ControlFile controlFile = AptPackageParser.parsePackage(tempBlob);
+        if (controlFile == null) {
+          throw new IOException("Invalid debian package:  no control file");
+        }
+        String name = controlFile.getField("Package").map(f -> f.value).get();
+        String version = controlFile.getField("Version").map(f -> f.value).get();
+        String architecture = controlFile.getField("Architecture").map(f -> f.value).get();
+        String assetPath = FacetHelper.buildAssetPath(name, version, architecture);
+        ensurePermitted(repository.getName(), AptFormat.NAME, assetPath, Collections.emptyMap());
```

Comment:
Instead of `Collections.emptyMap()` you should be able to pass a map of with name/version/arch, the idea being that they would be available for permissions though that isn't yet implemented for most formats.

e.g. we did this in the [NpmUploadHandler ](https://github.com/sonatype/nexus-public/blob/master/plugins/nexus-repository-npm/src/main/java/org/sonatype/nexus/repository/npm/NpmUploadHandler.java#L98)